Fix bug in `parse_as_chemfile`, `types` need to be reversed

rashidrafeek opened the pull request:
While using chemfiles to parse a structure, I noticed that the atom types where in reverse order in the output vtf file. Due to this the position and species pairing becomes incorrect in the output.

It seems this is due to the `types` vector being assembled in reverse order (`for i in Int(size(frame))-1:-1:0`). But all other properties are in normal order. Therefore the `types` vector need to be reversed. I have verified that this fixes the issue in the output vtf file.

Liozou commented:
Thank you for the PR! That makes sense yes, the reverse-assembly is necessary so we can do `Chemfiles.remove_atom!(frame, i)` safely but the `types` vector should indeed be reversed at the end.

Would it be possible to add a test file? I added this handling of non-cif files long ago but it is lacking tests at the moment. If possible, you could add an example file containing your structures in the `test/` folder, and add a line in `test/runtests.jl` that checks the topology of the file and compares it to what you expect it to find. It would look like
```
testcase = joinpath(last(_finddirs()), "test", "name_of_your_file") # the path to your file
@test extract1(determine_topology(testcase)) == "the_expected_output"
```
(if there are multiple intertwinned structures, you may want to remove the `extract1` call)

rashidrafeek commented:
Sure. I have added a test with an xyz file obtained from the already existing "Moganite.cif" file. Also, a test is added for the specific issue of ordering of types, which would fail in the current master. While testing, I noticed that there was an undefined variable in one of the error strings, which I have fixed now.

By the way, thanks for the great package.

Liozou commented:
Great, thank you for your contribution! That looks good, I'll wait for CI before merging.
